product.py

```diff
@@ -583,10 +583,9 @@ def findXenSourceProducts():
     installs = []
 
     for disk_device in diskutil.getQualifiedDiskList():
-        disk = diskutil.probeDisk(disk_device)
-
         inst = None
         try:
+            disk = diskutil.probeDisk(disk_device)
```

Comment:
Why is this change needed? Is there a new reason why `probeDisk()` can fail?

Reply:
OK, makes sense. There are multiple users of `probeDisk()`. Do the other users need to be adjusted too?